### PR TITLE
Hide on scrolling

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
fix https://github.com/nextcloud/android/issues/6238

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### Testing 
Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

[unit tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests)
[instrumented tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests)
[UI tests](https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests)

- [x] Tests written, or not not needed

### CODE REVIEW

File: src/main/java/com/owncloud/android/ui/activity/ToolbarActivity.java [hunk 1]
```diff
@@ -115,11 +119,15 @@ private void setupToolbar(boolean isHomeSearchToolbarShow) {
...
-        setupToolbar(false);
+        setupToolbar(false, false);
     }
 
     public void setupHomeSearchToolbar() {
-        setupToolbar(true);
+        setupToolbar(true, false);
+    }
+
+    public void setupHomeSearchToolbar(boolean showShortListGroup) {
```
Comment (JorisBodin) [inline src/main/java/com/owncloud/android/ui/activity/ToolbarActivity.java:129]: Rename `setupHomeSearchToolbar` by `setupHomeSearchToolbarWithShortListGroup`
Remove `showShortListGroup`

And add 
```
if (showSortListButtonGroup) {
    findViewById(R.id.sort_list_button_group).setVisibility(View.VISIBLE);
 }
```
After toolbar

I think it's more understandable that way.

**Comment by szaimen** [comment]
I have tested the PR and have found two issues:
1. The part below the header bar looks strange in dark mode: different grey tone?
2. The FAB button seems to need a long touch time to get hidden/shown. If I scroll fast (and thus touch scroll to trigger the scrolling), I can trigger the state where it doesn't get hidden/shown while scrolling.

**Comment by AndyScherzinger** [comment]
I can confirm both while
1. The toolbar and system bar have the wrong color and should have a lighter Grey tone
2. True while I'd be fine with making this a known issue for now

**Comment by tobiasKaminsky** [comment]
I created an issue for the two remaining things and will merge this PR after CI.

**Pull request merged**